types/react-table/index.d.ts

```diff
@@ -90,21 +96,26 @@ export type UseTableOptions<D extends object> = {
     getRowId: (originalRow: D, relativeIndex: number) => IdType<D>;
 }>;
 
-interface InstancePropGetter<D extends object> {
-    (props: object, instance: TableInstance<D>): object | Array<object | InstancePropGetter<D>>;
-}
+export type PropGetter<D extends object, Props, T extends any[] = [], P = Partial<Props>> =
+    | ((props: P, instance: TableInstance<D>, ...context: T) => P | Array<P>)
```

Comment:
To pass the lint, this needs to be:

```suggestion
    | ((props: P, instance: TableInstance<D>, ...context: T) => P | P[])
```